[FAL-774]: new Analytics module and ES/SQL improvements

eLRuLL opened the pull request:
This PR includes:

- "analytics" module, which setups every AWS resource [in this guide](https://github.com/open-craft/openedx-deployment/blob/master/docs/analytics/AWS_setup.md)
- Makes the ElasticSearch module more customizable
- Adds ability for the SQL module to include security groups to the RDS instance

**Reviewers:**
- [ ] @pomegranited

Commit:
FAL-774: makes elasticsearch module more customizable

Commit:
FAL-774: adds number_of_subnets to elasticsearch vpc_options

Commit:
FAL-774: adds possibility to determine when to start on subnets

Commit:
FAL-774: adds ability for RDS to add more security groups

Commit:
FAL-774: adds new analytics module

Commit:
FAL-774: adds analytics readme information and variables update

Commit:
FAL-774: analytics outputs the security group

Commit:
FAL-774: analytics outputs rds-emr security group

Commit:
FAL-875: adds RDS same access from master to replicas

pomegranited commented on modules/services/sql/variables.tf:
```diff
@@ -10,6 +10,10 @@ variable "max_allocated_storage" {
   default = 100
 }
 
+variable "extra_security_group_ids" {
```
Please add these new input variables to [sql/README](https://github.com/open-craft/terraform-scripts/blob/635890d9fb48661c6f531df494b1042bd23e5f1d/modules/services/sql/README.md#input)

eLRuLL replied:
done :+1:

pomegranited commented on modules/services/elasticsearch/variables.tf:
```diff
@@ -10,3 +10,26 @@ variable "number_of_nodes" {
   default = true
 }
+
+variable "zone_awareness_enabled" {
```
Please add these new input variables to [elasticsearch/README](https://github.com/open-craft/terraform-scripts/tree/635890d9fb48661c6f531df494b1042bd23e5f1d/modules/services/elasticsearch).

(Might be able to remove `number_of_subnets` and `start_number_of_subnets` depending on what you think about [this comment](https://github.com/open-craft/terraform-scripts/pull/18/files#diff-0f9cb3263848eba15ae3facdaa4b00ff2365de9c827c879aa4678d49855f6688R31)).

eLRuLL replied:
added :+1:

pomegranited commented on modules/services/elasticsearch/main.tf:
```diff
@@ -14,24 +14,21 @@ resource aws_elasticsearch_domain "openedx" {
-      tolist(data.aws_subnet_ids.default.ids)[1],
-      tolist(data.aws_subnet_ids.default.ids)[2],
-    ]
+    subnet_ids = slice(tolist(data.aws_subnet_ids.default.ids), var.start_number_of_subnets, var.number_of_subnets)
```
Are we ever going to use a subset of the `data.aws_subnet_ids` here? If so, then we're unlikely to be able to guarantee that the subnets we want are sequential in this list.

So I'd say either:
* Use all the subnets in `data.aws_subnet_ids`, OR
* Make the `subnet_ids` an input variable.

eLRuLL replied:
@pomegranited you are right :+1: I set all subnets by default and now you can specify the subnet IDs as well

pomegranited reviewed (changes requested):
@eLRuLL  This is looking great!
I haven't fully verified the analytics setup yet (since our client's analytics isn't yet running on their region), but the changes look good enough to merge, and we can iterate on the analytics stuff when we figure out what's not working there.

I've noted a few minor changes -- mostly docs, and one thing to simplify. But I'm good to approve once they're addressed.

Thank you for doing this! It's been needed for a long time.

Commit:
FAL-774: updates readmes and variables for ES and SQL

Commit:
FAL-774: better ElasticSearch subnets specification

Commit:
FAL-774: sets ES zone awareness to be dynamic

eLRuLL commented:
@pomegranited  this should be ready for you review again, thanks

pomegranited committed:
Grants access to ES from EMR security groups and analytics instance

and so adds the elasticsearch resource, to make configuring this resource
fully internal to the analytics module.

pomegranited commented:
:+1: Ready to squash and merge @eLRuLL , thank you for adding this!

- [x] I tested this:
   * Insights and the analytics tasks work using the resources created by this change.
   * Ran `terraform plan` for this client, and saw no changes needed. 
- [x] I read through the code
- [x] <strike>I checked for accessibility issues</strike>
- [x] Includes documentation
- [x] I made sure any change in configuration variables is reflected in the corresponding
    client's `configuration-secure` repository.
